contracts/cw4-group/src/contract.rs

```diff
@@ -207,6 +211,42 @@ fn list_members(
     Ok(MemberListResponse { members: members? })
 }
 
+fn list_members_by_weight(
+    deps: Deps,
+    start_after: Option<(u64, String)>,
+    limit: Option<u32>,
+) -> StdResult<MemberListResponse> {
+    let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+    // FIXME: Avoid building the (length-prefixed) bound manually
```

Comment:
Can't you use `.prefix((u64, String)).range(deps.storage, None, None, Order:Descending)`?

Ah, I see this is different. Definitely worth adding a helper function to storage-plus for this to encode from a key.
Maybe it is there?

```rust
  let start = start_after.map(|(w, a)| {
        let key = (UintKey64::from(w), &s).joined_key();
        Bound::exclusive(key)
     });
```

Reply:
Nice. Thanks for that. A slightly different version compiles:
```
    let start =
        start_after.map(|(w, a)| Bound::exclusive((U64Key::from(w), a.as_str()).joined_key()))```

Reply:
Maybe we could make this easier by exposing a method on `Map` (and all derivatives). Wait.. there is `Map.key(K)`. And we have a [Deref to &[u8] from Path](https://github.com/CosmWasm/cosmwasm-plus/blob/master/packages/storage-plus/src/path.rs#L20-L29). Maybe adding `Into<Vec<u8>>` would make this more efficient, but

`Bound::exclusive(&map.key((w.into(), &a)))` should probably work as well.

Feel free to add any helpers or docs as you see (and expose `key` on the various `Map` derivatives). We should make this easy.

Reply:
I tried using the `Map::key(K)` variant, but that requires a map of the proper key type.

I think it's OK now. We can always revisit / improve on this on another iteration.

Reply:
For indexed maps, I think we could add a `primary_key(K)` method that wraps `self.primary.key()`. That to differentiate it from the index keys.

Then, each index can have a `key()` method, that helps in building the index key. Not sure how to deal with generics here. We would probably need to add a `IK` trait to indexes. Need to think about it, play with it to know how to do it. Will open an issue for trying and doing this.

Reply:
Createdhttps://github.com/CosmWasm/cosmwasm-plus/issues/277 to track / implement this.